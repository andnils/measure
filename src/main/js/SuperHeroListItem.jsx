import React from 'react';
import PropTypes from 'prop-types';


export const SuperHeroListItem = (props) => {
  const { firstname, lastname } = props.hero;
  return <li>{`${firstname} ${lastname}`}</li>;
};


SuperHeroListItem.propTypes = {
  hero: PropTypes.shape({
    id: PropTypes.number,
    firstname: PropTypes.string,
    lastname: PropTypes.string,
    heroname: PropTypes.string
  })
};
