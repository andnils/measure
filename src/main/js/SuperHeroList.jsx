import React from 'react';
import PropTypes from 'prop-types';
import { SuperHeroListItem } from './SuperHeroListItem.jsx';


export class SuperHeroList extends React.Component {
  renderItems(heroes) {
    return heroes.map(hero => <SuperHeroListItem key={hero.id} hero={hero}/>);
  }

  render() {
    return (
      <div>
        <h1> Super heroez </h1>
        <ul>
          {this.renderItems(this.props.heroes)}
        </ul>
      </div>
    );
  }
}


SuperHeroList.propTypes = {
  heroes: PropTypes.arrayOf(PropTypes.object)
};

