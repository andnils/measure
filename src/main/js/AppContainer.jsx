import React from 'react';
import { SuperHeroList } from './SuperHeroList.jsx';


export class AppContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      heroes: []
    };
  }
  
  componentDidMount() {
    this.fetchHeroes();
  }

  fetchHeroes() {
    fetch('/api/heroes')
      .then((response) => response.ok ? response.json() : [])
      .then((data) => this.setState({ heroes: data }));
  }

  render() {
    const { heroes } = this.state;
    return (
      <div>
        <SuperHeroList heroes={heroes}/>
        { !heroes.length && <p> No data </p> }
      </div>
    );
  }
}

