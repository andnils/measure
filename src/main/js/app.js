import React from 'react';
import ReactDOM from 'react-dom';
import { AppContainer } from './AppContainer.jsx';

ReactDOM.render(
  React.createElement(AppContainer),
  document.getElementById('app')
);
