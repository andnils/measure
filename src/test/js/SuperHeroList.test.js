import React from 'react';
import renderer from 'react-test-renderer';
import { SuperHeroList } from '../../main/js/SuperHeroList';

const heroes = [
  { id: 1, firstname: 'Bruce', lastname: 'Banner', heroname: 'Hulk' },
  { id: 2, firstname: 'Peter', lastname: 'Parker', heroname: 'Spiderman' }
];

test('test if the super hero list renders same as before', () => {
  const tree = renderer
    .create(<SuperHeroList heroes={heroes}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


const sum = (a, b) => a + b;

test('adds 1 + 2 to equal 3', () => {
  expect(sum(1, 2)).toBe(3);
});
