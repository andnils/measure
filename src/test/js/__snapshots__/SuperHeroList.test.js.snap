// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if the super hero list renders same as before 1`] = `
<div>
  <h1>
     Super heroez 
  </h1>
  <ul>
    <li>
      Bruce Banner
    </li>
    <li>
      Peter Parker
    </li>
  </ul>
</div>
`;
